README.md

```diff
@@ -1,2 +1,151 @@
 # code-editor
 A code editor for tscircuit snippets that automatically loads types for snippets
+
+## Store Interface
+
+```typescript
+interface CodeEditorFile {
+  filename: string
+  content: string
+  language: 'typescript' | 'json' | 'javascript'
+}
+
+interface CodeEditorState {
+  // State
+  files: Record<string, CodeEditorFile>
+  activeFile: string
+  cursorPosition: number | null
+  isStreaming: boolean
+  readOnly: boolean
+  
+  // Config
+  showToolbar: boolean
+  baseApiUrl: string
+  codeCompletionApiKey?: string
+  
+  // File Actions
+  setFile: (filename: string, content: string) => void
+  updateFileContent: (filename: string, content: string) => void
+  setActiveFile: (filename: string) => void
+  removeFile: (filename: string) => void
+  
+  // Editor State
+  setCursorPosition: (position: number | null) => void
+  setIsStreaming: (isStreaming: boolean) => void
+  setReadOnly: (readOnly: boolean) => void
+  
+  // Editor Config
+  setShowToolbar: (show: boolean) => void
+  setBaseApiUrl: (url: string) => void
+  setCodeCompletionApiKey: (key: string) => void
+  
+  // Computed
+  dtsContent: string | null
+}
+
+// Events emitted by the editor
+interface CodeEditorEvents {
+  onCodeChange: (code: string, filename: string) => void
+  onCursorPositionChange: (position: number | null) => void
+  onDtsChange: (dts: string) => void
+  onError: (error: Error) => void
+}
+```
+
+## Component Props
+
+```typescript
+interface TscircuitCodeEditorProps {
+  // Optional configuration overrides
+  readOnly?: boolean
+  showToolbar?: boolean
+  baseApiUrl?: string
+  codeCompletionApiKey?: string
+  
+  // Event handlers
+  onCodeChange?: (code: string, filename: string) => void
+  onDtsChange?: (dts: string) => void
+  onError?: (error: Error) => void
+}
+```
+
+## Migration Guide
+
+1. Install the package:
+```bash
+npm install @tscircuit/code-editor
+```
+
+2. Replace existing CodeEditor implementation with TscircuitCodeEditor:
+
+```typescript
+// Before
+<CodeEditor
+  onCodeChange={handleCodeChange}
+  onDtsChange={handleDtsChange}
```

Comment:
```suggestion
```

We don't need to handle dts changes anymore, typescript is now bundled on server